articles/how-to-consume-data-from-an-API-using-react/index.md

```diff
@@ -0,0 +1,313 @@
+
+### Step 2: Creating a react project.
+
+We create a React application using the following `npx` command.
+
+```
+npx create-react-app omdapi
+```
+Once you execute the above statement, you have to give it a few minutes to enable the installation of the required dependencies. You can then open the folder in your preferred code editor when the installation is completed. Please note that to use `npx,` you must have `npm` installed.
+
+### Step 3: Understanding the react project structure
+
+All the files we will be working with are located in the `src` folder. These files will be compiled and shown to the user. The `public` folder stores all of the compiled files. The `App.js` helps to connect different `UI` components. The `index.css` stores the formatting properties of the body tag on the web page. The `App.css,` on the other hand, helps format individual components. The `node_modules` folder stores all the required React dependencies. The image below shows the overall structure of the React application.
+
+![structure](/engineering-education/how-to-consume-data-from-an-API-using-react/structure.png)
+
+### Step 4: Developing the react application
+
+Use `npm start ` to start the development server. All changes you make in the code editor are reflected in the application. By default, the react projects are usually hosted at `http://localhost:3000/`.
+Let’s delete all the unnecessary items in our project. Modify the `app.js` file to look like this.
+
+```javascript
+function App() {
+  return (
+    <div className="App">
+      // we will add our components here
+      <Movie/>
+    </div>
+  );
+}
+export default App;
+```
+
+Our app will have two major components: `Movie` page and `MovieItem`. The movie page will showcase the `movies` returned from the API, while the `MovieItem` presents precise information about the film. In other words, the `Movie` page hosts the `MovieItem`. Let’s create these items.
+
+### Step 5: Creating the Movie web page
+
+Please note that this page will hold our logic for requesting and handling data from the omdapi. To get started, create a new folder named components in the src directory. In the components folder, create a file called Movie.js. Before we move forward, we need to import `useEffect` and `useState`. We do so by adding the following line in the `Movie.js` file.
+
+```javascript
+import React, {useEffect, useState} from 'react';
+```
+
+The `useEffect` allows `React` to perform a specific action after a web page is loaded. In our case, we will use `useEffect` to make network requests to the API.
+`useState` method will help in state management. It ensures that `UI` components are updated in case there of any data changes. 
+
+The next step is to create a function named `Movie`, as shown below.
+
+```javascript
+Function Movie(){
+}
+```
+
+We will add other logic and elements in the above `Movie` component.
+
+We need to define several constants in the Movie component. 
```

Comment:
Also, it would be better if you instruct the users to add the HTML elements along with the associated state, like the input element instead of doing it far down below in another step.

The readers will have a better understanding of what is associated with what.